src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnWriter.Oid.cs

```diff
@@ -117,6 +117,12 @@ private void WriteObjectIdentifierCore(Asn1Tag tag, ReadOnlySpan<char> oidValue)
                 ReadOnlySpan<char> remaining = oidValue.Slice(2);
 
                 BigInteger subIdentifier = ParseSubIdentifier(ref remaining);
+
+                if (firstComponent <= 1 && subIdentifier >= 40)
```

Comment:
I think the comment a few lines up (right before the diff cuts off) explains why this is here.